commons-executors/src/main/java/com/palantir/common/concurrent/ExecutorInheritableThreadLocal.java

```diff
@@ -133,14 +156,14 @@ protected T initialValue() {
     /**
      * @return the old map installed on that thread
      */
-    static WeakHashMap<ExecutorInheritableThreadLocal<?>, Object> installMapOnThread(WeakHashMap<ExecutorInheritableThreadLocal<?>, Object> map) {
-        WeakHashMap<ExecutorInheritableThreadLocal<?>, Object> oldMap = mapForThisThread.get();
+    static ConcurrentMap<ExecutorInheritableThreadLocal<?>, Object> installMapOnThread(ConcurrentMap<ExecutorInheritableThreadLocal<?>, Object> map) {
+        ConcurrentMap<ExecutorInheritableThreadLocal<?>, Object> oldMap = perThreadThreadCache.get().asMap();
         if (map.isEmpty()) {
-            mapForThisThread.remove();
+            perThreadThreadCache.remove();
         } else {
             // Temporarily install the untransformed map in case callInstallOnChildThread makes use
             // of existing thread locals (UserSessionClientInfo does this).
-            mapForThisThread.set(map);
```

Comment:
Or we can just make the incoming map Immutable and that will expose who is setting map stuff during callInstallOnChildThread